lib/services/consumed/status-bar/status-bar.js

```diff
@@ -0,0 +1,41 @@
+/* @flow */
+
+import React from "react";
+import { Disposable } from "atom";
+
+import { StatusBar } from "./status-bar-component";
+import { SignalListView } from "./signal-list-view";
+
+import { reactFactory } from "../../../utils";
+
+import type { Store } from "../../../store";
+import type Kernel from "../../../kernel";
+
+export function addStatusBar(
+  store: Store,
+  statusBar: atom$StatusBar,
+  handleKernelCommand: Function
+) {
+  const statusBarElement = document.createElement("div");
+  statusBarElement.classList.add("inline-block", "hydrogen");
+
+  const statusBarTile = statusBar.addLeftTile({
+    item: statusBarElement,
+    priority: 100
+  });
+
+  const onClick = showKernelCommands.bind(this, handleKernelCommand);
+
+  reactFactory(<StatusBar store={store} onClick={onClick} />, statusBarElement);
+
+  return new Disposable(() => statusBarTile.destroy());
+}
+
+function showKernelCommands(handleKernelCommand, kernel) {
+  if (!this.signalListView) {
+    this.signalListView = new SignalListView(kernel);
```

Comment:
Why should `SignalListView`  turn to be constructed from `kernel` ?
Does this work with multiple kernels ?

Reply:
It does work for multiple kernels, and we should be passing more arguments than just importing store. It makes for easier test cases and debugs. It also only provides the necessary data for the task. You don't need the whole store. Also I recommend you try testing this because it works with multiple kernels in my tests.

Reply:
I see. Then changing the kernel by `onClick` handler would be needed.